src/storage/btree_load.c

```diff
@@ -4927,33 +4929,37 @@ online_index_builder (THREAD_ENTRY * thread_p, BTID_INT * btid_int, HFID * hfids
 	}
     }
 
-  /* Check if the workerpool is empty */
-  do
+  /* Check if the workerpool is empty only if we did not get any other errors. */
+  if (ret == NO_ERROR)
     {
-      bool dummy_continue_checking = true;
-
-      if (load_context.m_has_error != NO_ERROR)
+      do
 	{
-	  /* Also stop all threads. */
-	  thread_get_manager ()->destroy_worker_pool (ib_workpool);
-	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, load_context.m_error_code, 0);
-	  return load_context.m_error_code;
-	}
+	  bool dummy_continue_checking = true;
+
+	  if (load_context.m_has_error != NO_ERROR)
+	    {
+	      /* Also stop all threads. */
+	      thread_get_manager ()->destroy_worker_pool (ib_workpool);
```

Comment:
It seems easier to break on errors and remove 'assert (load_context.m_tasks_executed == tasks_started);' safe-guard after while loop.